### DISCUSSION TITLE
CIP communication fails after firmware upgrade

### DISCUSSION THREAD

**Discussion by WimVriend** [digitalpetri/ethernet-ip#62] [General]
My colleague has upgraded the firmware of our 1756-L85E CPU from v33 to v35. After doing this, the CIP-client is unable to connect to the PLC, so no data can be read (pool.aquire fails).

It is however possible to retrieve the firmware version of the CPU, using the GetAttributeListService. 

Could this be related to the implementation of the CIP client, or can it be some issue related to licensing or something like that?

**Comment by kevinherron** [comment]
What's the error message / code? Do you have a capture you can share?

**Reply by WimVriend** [reply]
![Error after updating firmware](https://github.com/digitalpetri/ethernet-ip/assets/46961736/26dbb9e4-16be-4d55-b9f6-29d144c38587)

Hello Kevin, this is the screenshot my colleague supplied, with the error we receive after upgrading the firmware of the CPU.

**Reply by WimVriend** [reply]
It would be great if you can make sense of this, because I think this is an accident waiting to happen...

**Reply by kevinherron** [reply]
This means "Invalid Segment in Connection Path":
<img width="772" alt="image" src="https://github.com/digitalpetri/ethernet-ip/assets/340273/9f2de319-b8de-4cd3-b6f1-6329d0726d9e">

What does your configured connection path look like? Can you get a Wireshark capture of this?

We upgraded one of our controllers to v35 when you first reported this and there was no issue.

**Reply by WimVriend** [reply]
@kevinherron : I sent you an e-mail, containing two Wireshark captures. Any help is greatly appreciated ...

**Reply by kevinherron** [reply]
@WimVriend does it work with v35 if you use an "empty" connection path like this instead?
```java
var connectionPath = new PaddedEPath(List.of());
```

**Reply by Marc-Dierikx** [reply]
As the mentioned colleague of Wim, I can confirm that with that setting the connection succeeds. Tags addressed  in the same manner as with the 33 version firmware are not recognized though.

**Reply by kevinherron** [reply]
@Marc-Dierikx  I noticed in the Wireshark captures that you are not configuring a vendor ID, which results in the ForwardOpen request using 0 for originator vendor ID. This is the only difference I can see when compared to the Kepware capture, which sets the vendor ID to the one assigned to their org.

Can you try setting a non-zero vendor ID and see if that changes anything?

**Reply by Marc-Dierikx** [reply]
That does the trick. Thanks!

**Reply by kevinherron** [reply]
That explains why we've never seen this issue in our usage and weren't able to reproduce.

**Comment by WimVriend** [comment]
At this moment I don't have the exact message, since my colleague did not save it and now he has downgraded the CPU to v34 (which does work, so both v33 and v34 work, only v35 does not).

I will have to ask him if he wants to go through the motions again next week ;)
